From 1a616ef74b66a45c1e16ec12e46fabfc03613668 Mon Sep 17 00:00:00 2001
From: Mohammed Gamal <mgamal@redhat.com>
Date: Thu, 17 Nov 2022 18:58:31 +0100
Subject: [PATCH 08/14] tools: hv: kvp: remove unnecessary (void*) conversions

RH-Author: Miroslav Rezanina <mrezanin@redhat.com>
RH-MergeRequest: 9: Synchronize RHEL 9 changes to RHEL 10
RH-Jira: RHEL-40107 RHEL-40679
RH-Acked-by: Ani Sinha <anisinha@redhat.com>
RH-Commit: [8/14] 16a5e8f5c4799f6777600cd62b4621e450638f22 (mrezanin/centos-git-hyperv-daemons)

commit 2258954234db7530e9d86bb32cd6ad54485ff926
Author: Zhou jie <zhoujie@nfschina.com>
Date:   Tue Aug 23 11:45:52 2022 +0800

    tools: hv: kvp: remove unnecessary (void*) conversions

    Remove unnecessary void* type casting.

    Signed-off-by: Zhou jie <zhoujie@nfschina.com>
    Reviewed-by: Michael Kelley <mikelley@microsoft.com>
    Link: https://lore.kernel.org/r/20220823034552.8596-1-zhoujie@nfschina.com
    Signed-off-by: Wei Liu <wei.liu@kernel.org>

Signed-off-by: Mohammed Gamal <mgamal@redhat.com>

patch_name: hpvd-tools-hv-kvp-remove-unnecessary-void-conversions.patch
present_in_specfile: true
location_in_specfile: 11
---
 hv_kvp_daemon.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hv_kvp_daemon.c b/hv_kvp_daemon.c
index c97c12e..27f5e7d 100644
--- a/hv_kvp_daemon.c
+++ b/hv_kvp_daemon.c
@@ -772,11 +772,11 @@ static int kvp_process_ip_address(void *addrp,
 	const char *str;
 
 	if (family == AF_INET) {
-		addr = (struct sockaddr_in *)addrp;
+		addr = addrp;
 		str = inet_ntop(family, &addr->sin_addr, tmp, 50);
 		addr_length = INET_ADDRSTRLEN;
 	} else {
-		addr6 = (struct sockaddr_in6 *)addrp;
+		addr6 = addrp;
 		str = inet_ntop(family, &addr6->sin6_addr.s6_addr, tmp, 50);
 		addr_length = INET6_ADDRSTRLEN;
 	}
-- 
2.39.3

